plugins/languages/rust/rustaceanvim/default.nix

```diff
@@ -82,9 +80,9 @@ helpers.neovim-plugin.mkNeovimPlugin config {
 
         # Ensure the plugin config is placed **after** the rest of the LSP configuration
         # (and thus after the declaration of `__lspOnAttach`)
-        plugins.lsp.postConfig = configStr;
+        plugins.lsp.postConfig = cfg.config.final;
       })
       # Else, just put the plugin config anywhere
-      (mkIf (!config.plugins.lsp.enable) { extraConfigLua = configStr; })
+      (mkIf (!config.plugins.lsp.enable) { extraConfigLua = cfg.config.final; })
```

Comment:
Is this why you gated the `final` -> `configLocation` assignment behind `callSetup`? If so I think we should introduce a different way to do that.

Perhaps `configLocation` could be nullable and setting it to null prevents assigning anything to it?